File: wrappers/java/src/test/java/org/hyperledger/indy/sdk/signus/GetDidMetadataTest.java
package org.hyperledger.indy.sdk.signus;

import org.hyperledger.indy.sdk.IndyIntegrationTestWithSingleWallet;
import org.hyperledger.indy.sdk.wallet.WalletValueNotFoundException;
import org.junit.Test;

import java.util.concurrent.ExecutionException;

import static org.hamcrest.CoreMatchers.isA;
import static org.junit.Assert.assertEquals;


public class GetDidMetadataTest extends IndyIntegrationTestWithSingleWallet {

	@Test
	public void testGetDidMetadataWorks() throws Exception {
		Signus.setDidMetadata(wallet, DID, METADATA).get();
		String receivedMetadata = Signus.getDidMetadata(wallet, DID).get();
		assertEquals(METADATA, receivedMetadata);
	}

	@Test
	public void testGetDidMetadataWorksForEmptyString() throws Exception {
		Signus.setDidMetadata(wallet, DID, "").get();
		String receivedMetadata = Signus.getDidMetadata(wallet, DID).get();
		assertEquals("", receivedMetadata);
	}

	@Test
	public void testGetDidMetadataWorksForNoMetadata() throws Exception {
		thrown.expect(ExecutionException.class);
		thrown.expectCause(isA(WalletValueNotFoundException.class));

		Signus.getDidMetadata(wallet, DID).get();
	}
}

File: wrappers/dotnet/indy-sdk-dotnet-test/SignusTests/GetDidMetadataTest.cs
using Hyperledger.Indy.SignusApi;
using Hyperledger.Indy.WalletApi;
using Microsoft.VisualStudio.TestTools.UnitTesting;
using System.Threading.Tasks;

namespace Hyperledger.Indy.Test.SignusTests
{
    [TestClass]
    public class GetDidMetadataTest : IndyIntegrationTestWithSingleWallet
    {
        [TestMethod]
        public async Task TestGetDidMetadataWorks()
        {
            await Signus.SetDidMetadataAsync(wallet, DID1, METADATA);
            var receivedMetadata = await Signus.GetDidMetadataAsync(wallet, DID1);

            Assert.AreEqual(METADATA, receivedMetadata);
        }

        [TestMethod]
        public async Task TestGetDidMetadataWorksForEmptyString()
        {
            await Signus.SetDidMetadataAsync(wallet, DID1, string.Empty);
            var receivedMetadata = await Signus.GetDidMetadataAsync(wallet, DID1);

            Assert.AreEqual(string.Empty, receivedMetadata);
        }

        [TestMethod]
        public async Task TestGetDidMetadataWorksForNoMetadata()
        {
            var ex = await Assert.ThrowsExceptionAsync<WalletValueNotFoundException>(() =>
               Signus.GetDidMetadataAsync(wallet, DID1)
           );
        }
    }
}